3-extra/1-count-words.js

```diff
@@ -25,8 +25,21 @@
 
 function countWords(string) {
   const wordCount = {};
+  // if (string.length){
+  //   string.split("").forEach(element =>
+  //     wordCount.sameProperty(element)? wordCount[element] += 1: wordCount[element] =1);
+  //   }
 
   // write code here
+  if (string.length) {
+    string
+      .split(" ")
+      .forEach((word) =>
+        wordCount.hasOwnProperty(word)
```

Comment:
works here, however I'd advise keeping it simpler by doing something like `wordCount[word] != null ? wordCount[count] +=1 : wordCount[count] = 1` - hasOwnProperty might cause issues when you start using things like `class`es and more complicated objects. the `x != null` check will check for `null` OR `undefined` (instead of `!wordCount[word]` which would check for falsey values, of which `0` is one).

Reply:
![image](https://user-images.githubusercontent.com/49273611/211012811-0d5b1268-ba75-4088-b99e-83edf513f0e4.png)
also, `.hasOwnProperty` doesn't actually check the value of the property, leading you to situations like this where even though the `name` property is actually set to undefined, the check passes because `undefined` is a valid value in JS.